spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -245,6 +251,22 @@ public static Expression convert(Predicate predicate) {
     return null;
   }
 
+  private static <T> Pair<UnboundTerm<T>, T> predicateChildren(Predicate predicate) {
```

Comment:
Question: should we remove <T>?  Looks like all callers just use Object in the end, and <T> just clutters a bit?

Reply:
@szehon-ho It should be OK to remove the `<T>` and use `Object` instead. And also updated `handleEqual` and `handleNotEqual`.